app/src/main/java/fr/neamar/kiss/loader/LoadContactsPojos.java

```diff
@@ -53,8 +54,24 @@ public LoadContactsPojos(Context context) {
                             .getColumnIndex(ContactsContract.Contacts.LOOKUP_KEY));
                     contact.timesContacted = Integer.parseInt(cur.getString(cur
                             .getColumnIndex(ContactsContract.CommonDataKinds.Phone.TIMES_CONTACTED)));
-                    contact.setName(cur.getString(cur
-                            .getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+                    contact.setName(cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME)));
+
+                    String contactID = cur.getString(cur.getColumnIndex(ContactsContract.CommonDataKinds.Phone.CONTACT_ID));
+
+                    Cursor cur2 = context.getContentResolver().query(
```

Comment:
I was unable to execute a raw query, but I implemented it in a slightly different way that should bring a performance improvement comparable to that of a JOIN (and it's also a lot cleaner).